pkg/cache/snapshot.go

```diff
@@ -103,10 +105,27 @@ func (s *Snapshot) Log(log logr.Logger) {
 	}
 }
 
-func (c *Cache) Snapshot(ctx context.Context) (*Snapshot, error) {
+type snapshotOption struct {
+	penalties *utilmaps.SyncMap[utilqueue.LocalQueueReference, corev1.ResourceList]
+}
+
+type SnapshotOption func(*snapshotOption)
+
+func WithPenalties(penalties *utilmaps.SyncMap[utilqueue.LocalQueueReference, corev1.ResourceList]) SnapshotOption {
```

Comment:
```suggestion
func WithAfsEntryPenalties(penalties *utilmaps.SyncMap[utilqueue.LocalQueueReference, corev1.ResourceList]) SnapshotOption {
```
since this is used outside of the AFS packages, just "panelties" it might be a bit cryptic

Reply:
Done.